### PR TITLE
add GetErrorMessage of types.receipt

### PR DESCRIPTION
**Pull request by sulenn**
parse exception output value for channel protocol if the receipt.Status is not equal 0x0

### CODE REVIEW

File: conn/channel.go [hunk 1]
```diff
@@ -456,9 +457,16 @@ func (hc *channelSession) sendTransaction(ctx context.Context, msg interface{})
 		return nil, fmt.Errorf("parse receipt error %w", err)
 	}
 	if transactionReceipt.Status != "0x0" {
-		return nil, fmt.Errorf("receipt error code:%s", transactionReceipt.Status)
+		if len(transactionReceipt.Output) >= 138 { // 0x + 4 bytes funcName + 32 bytes offset + 32 bytes string length
```
Comment (bxq2011hust) [inline conn/channel.go:460]: （4+32+32）*2 = 136

**Comment by fisco-dev** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- precompiled/config/system_config_service.go  2
- core/types/receipt.go  27
- precompiled/cns/cns_service.go  2
- precompiled/crud/crud_service.go  2
         

Complexity decreasing per file
==============================
+ core/types/status_code.go  -13
         

Clones added
============
- precompiled/config/system_config_service.go  1
- precompiled/crud/crud_service.go  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/FISCO-BCOS/go-sdk/pullRequest?prid=5904758&bid=19336934)

File: core/types/receipt.go [hunk 2]
```diff
@@ -87,6 +186,29 @@ func (r *Receipt) GetOutput() string {
...
+	if status == Success {
+		return "", nil
+	}
+	errorMessage = getStatusMessage(status)
+	if len(r.Output) >= 138 { // 0x + 4 bytes funcName + 32 bytes offset + 32 bytes string length
+		outputBytes, err := hex.DecodeString(r.Output[2:])
+		if err != nil {
+			return "", fmt.Errorf("ParseErrorMessage failed, hex.DecodeString err: %v", err)
+		}
+		errorMessage += ", " + string(outputBytes[68:])
```
Comment (bxq2011hust) [inline core/types/receipt.go:206]: utf8？

File: core/types/receipt.go [hunk 3]
```diff
@@ -87,6 +186,29 @@ func (r *Receipt) GetOutput() string {
 	return r.Output
 }
 
+// ParseErrorMessage parses unusual output value from Receipt
+func (r *Receipt) ParseErrorMessage() (string, error) {
```
Comment (bxq2011hust) [inline core/types/receipt.go:190]:
```suggestion
func (r *Receipt) GetErrorMessage() (string, error) {
```

File: core/types/receipt.go [hunk 4]
```diff
@@ -62,9 +158,12 @@ func (r *Receipt) GetRoot() string {
 }
 
 // GetStatus returns the transaction status
-func (r *Receipt) GetStatus() int {
-	status, _ := strconv.ParseInt(r.Status, 16, 32)
-	return int(status)
+func (r *Receipt) GetStatus() (int, error) {
```
Comment (bxq2011hust) [inline core/types/receipt.go:161]:
```suggestion
func (r *Receipt) GetStatus() (int) {
```

**Comment by fisco-dev** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- precompiled/crud/crud_service.go  1
- precompiled/cns/cns_service.go  1
- precompiled/config/system_config_service.go  1
- core/types/receipt.go  27
         

Complexity decreasing per file
==============================
+ core/types/status_code.go  -13
         

Clones added
============
- precompiled/crud/crud_service.go  1
- precompiled/config/system_config_service.go  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/FISCO-BCOS/go-sdk/pullRequest?prid=5904758&bid=19336934)

File: core/types/receipt.go (hunk 2 above)
Comment (sulenn) [reply core/types/receipt.go:203]: string()，default is utf8

**Comment by fisco-dev** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- core/types/receipt.go  27
         

Complexity decreasing per file
==============================
+ core/types/status_code.go  -13
         

Clones removed
==============
+ precompiled/contractlifecycle/contract_life_cycle_service.go  -1
+ precompiled/cns/cns_service.go  -1
+ precompiled/consensus/consensus_service.go  -1
+ precompiled/permission/permission_service.go  -1
+ precompiled/config/system_config_service.go  -1
+ precompiled/chaingovernance/chain_governance_service.go  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/FISCO-BCOS/go-sdk/pullRequest?prid=5904758&bid=19336934)

**Pull request merged**